circuit/types/group/src/lib.rs

```diff
@@ -129,6 +129,18 @@ impl<E: Environment> Group<E> {
         // i.e. that it is 4 (= cofactor) times the postulated point on the curve.
         double_point.enforce_double(self);
     }
+
+    /// Returns a `Boolean` indicating if `self` is in the largest prime-order subgroup.
+    pub fn is_in_group(&self) -> Boolean<E> {
```

Comment:
I suggest adding here some of the comments about the code that was moved here, namely that we can perform scalar multiplication also on (1) points not in the subgroup (the point may not be in the subgroup) and (2) scalars not in the scalar type (the order does not have scalar type). It wouldn't necessarily be the case for this to be possible, if the group type and operations were defined more restrictively.

Reply:
Added some comments, please advise if additional ones are needed for clarity.